samples/exportDataToBigquery.js

```diff
@@ -0,0 +1,69 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+'use strict';
+
+function main(projectId, bigqueryProjectId, bigqueryDataset, bigqueryTable) {
+  // [START contactcenterinsights_export_data_to_bigquery]
+  /**
+   * TODO(developer): Uncomment these variables before running the sample.
+   */
+  // const projectId = 'my_project_id';
+  // const bigqueryProjectId = 'my_bigquery_project_id';
+  // const bigqueryDataset = 'my_bigquery_dataset';
+  // const bigqueryTable = 'my_bigquery_table';
+
+  // Imports the Contact Center Insights client.
+  const {
+    ContactCenterInsightsClient,
+  } = require('@google-cloud/contact-center-insights');
+
+  // Instantiates a client.
+  const client = new ContactCenterInsightsClient();
+
+  async function exportDataToBigquery() {
+    const [operation] = await client.exportInsightsData({
+      parent: client.locationPath(projectId, 'us-central1'),
+      bigQueryDestination: {
+        projectId: bigqueryProjectId,
+        dataset: bigqueryDataset,
+        table: bigqueryTable,
+      },
+    });
+
+    // Wait for operation to complete.
+    try {
+      await operation.promise();
+    } catch (e) {
+      if (
+        e.message ===
```

Comment:
This feels like a strange corner case to make customers handle.  @alexander-fenster is this expected to be a normal thing?  What are the situations where this could happen?

Reply:
I can try to give some context. This is a normal and expected behavior from the Contact Center Insights API. Each time an export operation finishes successfully, it'll return an empty response[1]. The response message is empty by design. If the export operation fails, it'll return an error and it won't match this particular message that we're trying to catch.

I'm not sure why the NodeJS Operation library throws an error when it encounters an empty response (please see line 275 of [2] for this particular error message that we're trying to catch). Ideally NodeJS shouldn't throw an error. But since it does, I want to convey in this sample that "if the API returns this particular error message, it means the export operation finishes successfully", and hence I wrote this `try catch` statement. Let me know how I should update the code to better convey this message!

[1] https://cloud.google.com/contact-center/insights/docs/reference/rpc/google.cloud.contactcenterinsights.v1#exportinsightsdataresponse
[2] https://googleapis.github.io/gax-nodejs/classes/Operation.html#source

Reply:
https://github.com/googleapis/gax-nodejs/pull/1087 will fix it. Please, if you see some weirdness that makes you do things like this, escalate to us (@summer-ji-eng or me), it's better to fix it once and forever than let customers copy-paste the workarounds :)

Reply:
Ack! Thank you for the quick fix. I tested my PR with your fix and found that the code ran successfully without the weird `try catch` (I also removed this `try catch`). Thanks again for your help!